### ISSUE TITLE
Massive slowdown between version 5.19.2 and 5.19.3 (and following)

### ISSUE THREAD

**Issue by alxsimon** [snakemake/snakemake#568]
I recently wanted to update snakemake.
I was using v5.19.2 before.
Snakemake is installed through a conda environment on Ubuntu server 18.04

Issue: there is a massive slowdown in building the dag of jobs from this version up (from seconds to minutes).

Trying several version, I could pinpoint the change between 5.19.2 and 5.19.3

v5.19.2
```
snakemake --use-singularity --use-conda --conda-prefix .conda -j 16 -U gs_fit -n 
0,39s user 0,04s system 98% cpu 0,435 total                                                                 
```
v5.19.3
```
snakemake --use-singularity --use-conda --conda-prefix .conda -j 16 -U gs_fit -n 
2,27s user 1,84s system 1% cpu 4:36,90 total                                                                   
```

I find the same behaviour in other pipelines I have, so it does not seem to be pipeline specific.
Snakemake is installed alone in its conda environment.

**Comment by vsoch** [comment]
hey @alxsimon - could it be the generation of a local cache (called a file inventory) e.g., here? https://github.com/snakemake/snakemake/commit/01cf67fd54f8c86e61a26da1d9b44518414ef20a#diff-7861b1b7ac192ee3dd0e12211b6d030dR772. I would first try adding a timer around that call to see how long it takes. If it's not the issue, then we can look elsewhere.

**Comment by alxsimon** [comment]
Hi @vsoch, it seems indeed to be the culprit.
A timer around the `file.inventory()` call produced at least on the first call a long delay:
```
320.41467928886414
1.1444091796875e-05
1.0967254638671875e-05
1.0251998901367188e-05
1.0013580322265625e-05
0.17336630821228027
4.5299530029296875e-06
3.0994415283203125e-06
3.337860107421875e-06
3.0994415283203125e-06
...[a lot of other e-06 values]
```

I imagine this file caching could be slowed down if there are mounted sshfs folders in the path.
Unfortunately I cannot go around the use of sshfs right know as I don't have the local storage space necessary.
Could there be an option for not doing this inventory?
Cheers

**Comment by vsoch** [comment]
Glad we found the culprit! @johanneskoester what are your thoughts on having an ability to skip the inventory?

**Comment by alxsimon** [comment]
Solved by 5.26.0
